File: apps/desktop/src-tauri/src/sys/commands/analytics.rs
use crate::sys::telemetry::{
    AnalyticsMetricsCollector, AppMetrics, SystemMetrics, TelemetryCollector, TelemetryEvent,
};
use chrono;
use serde_json::Value;
use std::collections::HashMap;
use std::sync::Arc;
use tauri::State;
use tokio::sync::RwLock;

pub struct TelemetryState {
    pub collector: Arc<RwLock<TelemetryCollector>>,
    pub metrics_collector: Arc<RwLock<AnalyticsMetricsCollector>>,
}

impl TelemetryState {
    pub fn new(
        collector: TelemetryCollector,
        metrics_collector: AnalyticsMetricsCollector,
    ) -> Self {
        Self {
            collector: Arc::new(RwLock::new(collector)),
            metrics_collector: Arc::new(RwLock::new(metrics_collector)),
        }
    }
}

#[tauri::command]
pub async fn analytics_track_event(
    event: TelemetryEvent,
    state: State<'_, TelemetryState>,
) -> Result<(), String> {
    let collector = state.collector.read().await;

    if !collector.is_enabled() {
        return Ok(());
    }

    drop(collector);
    let collector = state.collector.write().await;

    collector
        .track(event)
        .await
        .map_err(|e| format!("Failed to track event: {}", e))
}

#[tauri::command]
pub async fn analytics_flush_events(state: State<'_, TelemetryState>) -> Result<(), String> {
    let collector = state.collector.read().await;
    collector
        .flush()
        .await
        .map_err(|e| format!("Failed to flush events: {}", e))
}

#[tauri::command]
pub async fn analytics_get_session_id(state: State<'_, TelemetryState>) -> Result<String, String> {
    let collector = state.collector.read().await;
    Ok(collector.get_session_id())
}

#[tauri::command]
pub async fn analytics_set_user_property(
    key: String,
    value: Value,
    state: State<'_, TelemetryState>,
) -> Result<(), String> {
    let collector = state.collector.read().await;
    collector
        .set_user_property(key, value)
        .await
        .map_err(|e| format!("Failed to set user property: {}", e))
}

#[tauri::command]
pub async fn metrics_get_system(state: State<'_, TelemetryState>) -> Result<SystemMetrics, String> {
    let mut collector = state.metrics_collector.write().await;
    Ok(collector.collect_system_metrics())
}

#[tauri::command]
pub async fn metrics_get_app(state: State<'_, TelemetryState>) -> Result<AppMetrics, String> {
    let collector = state.metrics_collector.read().await;
    Ok(collector.collect_app_metrics())
}

use crate::sys::commands::settings::SettingsState;

#[tauri::command]
pub async fn feature_flag_get(
    flag_name: String,
    settings_state: State<'_, SettingsState>,
) -> Result<bool, String> {
    get_feature_flag(&flag_name, &settings_state).await
}

pub async fn get_feature_flag(
    flag_name: &str,
    settings_state: &SettingsState,
) -> Result<bool, String> {
    let settings = settings_state.settings.lock().await;
    Ok(settings
        .feature_flags
        .get(flag_name)
        .copied()
        .unwrap_or(false))
}

#[tauri::command]
pub async fn feature_flag_get_all(
    settings_state: State<'_, SettingsState>,
) -> Result<HashMap<String, bool>, String> {
    get_all_feature_flags(&settings_state).await
}

pub async fn get_all_feature_flags(
    settings_state: &SettingsState,
) -> Result<HashMap<String, bool>, String> {
    let settings = settings_state.settings.lock().await;
    Ok(settings.feature_flags.clone())
}

#[tauri::command]
pub async fn analytics_delete_all_data(state: State<'_, TelemetryState>) -> Result<(), String> {
    let collector = state.collector.read().await;
    collector
        .delete_all_data()
        .await
        .map_err(|e| format!("Failed to delete analytics data: {}", e))
}

#[tauri::command]
pub async fn metrics_increment_automations(state: State<'_, TelemetryState>) -> Result<(), String> {
    let mut collector = state.metrics_collector.write().await;
    collector.increment_automations_count();
    Ok(())
}

#[tauri::command]
pub async fn metrics_increment_goals(state: State<'_, TelemetryState>) -> Result<(), String> {
    let mut collector = state.metrics_collector.write().await;
    collector.increment_goals_count();
    Ok(())
}

#[tauri::command]
pub async fn metrics_set_mcp_servers(
    count: u64,
    state: State<'_, TelemetryState>,
) -> Result<(), String> {
    let mut collector = state.metrics_collector.write().await;
    collector.set_mcp_servers_count(count);
    Ok(())
}

#[tauri::command]
pub async fn metrics_set_cache_hit_rate(
    rate: f64,
    state: State<'_, TelemetryState>,
) -> Result<(), String> {
    let mut collector = state.metrics_collector.write().await;
    collector.set_cache_hit_rate(rate);
    Ok(())
}

#[tauri::command]
pub async fn analytics_get_usage_stats(
    state: State<'_, AppDatabase>,
) -> Result<serde_json::Value, String> {
    let db = create_analytics_db_connection(&state)?;
    let conn = db.lock().await;

    let total_events: i64 = conn
        .query_row("SELECT COUNT(*) FROM automation_history", [], |row| {
            row.get(0)
        })
        .unwrap_or(0);

    let today_start = chrono::Utc::now()
        .date_naive()
        .and_hms_opt(0, 0, 0)
        .map(|t| t.and_utc().timestamp())
        .unwrap_or(0);
    let events_today: i64 = conn
        .query_row(
            "SELECT COUNT(*) FROM automation_history WHERE created_at >= ?1",
            [today_start],
            |row| row.get(0),
        )
        .unwrap_or(0);

    let avg_session_duration_ms: f64 = conn.query_row(
        "SELECT AVG(last_activity - started_at) * 1000 FROM user_sessions WHERE last_activity > started_at",
        [],
        |row| row.get(0),
    ).unwrap_or(0.0);

    let active_days: i64 = conn
        .query_row(
            "SELECT COUNT(DISTINCT date(started_at, 'unixepoch')) FROM user_sessions",
            [],
            |row| row.get(0),
        )
        .unwrap_or(0);

    let mau: i64 = conn
        .query_row(
            "SELECT COUNT(DISTINCT date(started_at, 'unixepoch')) FROM user_sessions WHERE started_at >= ?1",
            [chrono::Utc::now().timestamp() - 30 * 24 * 3600],
            |row| row.get(0),
        )
        .unwrap_or(0);

    let new_users_today: i64 = conn
        .query_row(
            "SELECT COUNT(*) FROM user_sessions WHERE started_at >= ?1 AND id NOT IN (SELECT id FROM user_sessions WHERE started_at < ?1)",
            [today_start],
            |row| row.get(0),
        ).unwrap_or(0);

    Ok(serde_json::json!({
        "dau": if events_today > 0 { 1 } else { 0 },
        "mau": mau,
        "total_users": 1,
        "new_users_today": new_users_today,
        "new_users_this_week": 0,
        "new_users_this_month": 0,
        "avg_session_duration_ms": avg_session_duration_ms,
        "total_events": total_events,
        "events_today": events_today,
        "retention_rate": if active_days > 0 { 100.0 } else { 0.0 },
    }))
}

#[tauri::command]
pub async fn analytics_get_feature_usage(
    state: State<'_, AppDatabase>,
) -> Result<Vec<serde_json::Value>, String> {
    let db = create_analytics_db_connection(&state)?;
    let conn = db.lock().await;

    let mut stmt = conn
        .prepare(
            "SELECT task_type, COUNT(*) as usage_count, MAX(created_at) as last_used
         FROM automation_history
         GROUP BY task_type
         ORDER BY usage_count DESC",
        )
        .map_err(|e| e.to_string())?;

    let features = stmt
        .query_map([], |row| {
            let feature_name: String = row.get(0)?;
            let usage_count: i64 = row.get(1)?;
            let last_used: Option<String> = row.get(2)?;

            Ok(serde_json::json!({
                "feature_name": feature_name,
                "usage_count": usage_count,
                "unique_users": 1,
                "trend": "stable",
                "last_used": last_used
            }))
        })
        .map_err(|e| e.to_string())?
        .collect::<Result<Vec<_>, _>>()
        .map_err(|e| e.to_string())?;

    Ok(features)
}

use crate::data::analytics::{
    MetricsAggregator, ProcessMetrics, ROICalculator, ROIReport, ReportGenerator,
    ScheduledReportGenerator, ToolMetrics, TrendPoint, UserMetrics,
};
use crate::sys::commands::AppDatabase;
use rusqlite::Connection;

fn create_analytics_db_connection(
    app_db: &AppDatabase,
) -> Result<Arc<tokio::sync::Mutex<Connection>>, String> {
    let _conn = app_db
        .conn
        .lock()
        .map_err(|e| format!("Failed to lock database: {}", e))?;

    let db_path = std::env::var("AGI_DB_PATH").unwrap_or_else(|_| {
        std::path::PathBuf::from(".")
            .join("agiworkforce.db")
            .to_string_lossy()
            .to_string()
    });

    let new_conn = Connection::open(&db_path)
        .map_err(|e| format!("Failed to open analytics connection: {}", e))?;

    Ok(Arc::new(tokio::sync::Mutex::new(new_conn)))
}

#[tauri::command]
pub async fn analytics_calculate_roi(
    start_date: i64,
    end_date: i64,
    state: State<'_, AppDatabase>,
) -> Result<ROIReport, String> {
    let db = create_analytics_db_connection(&state)?;
    let calculator = ROICalculator::new(db);

    calculator
        .calculate_roi(start_date, end_date)
        .await
        .map_err(|e| format!("Failed to calculate ROI: {}", e))
}

#[tauri::command]
pub async fn analytics_get_process_metrics(
    start_date: i64,
    end_date: i64,
    state: State<'_, AppDatabase>,
) -> Result<Vec<ProcessMetrics>, String> {
    let db = create_analytics_db_connection(&state)?;
    let aggregator = MetricsAggregator::new(db);

    aggregator
        .aggregate_by_process_type(start_date, end_date)
        .await
        .map_err(|e| format!("Failed to aggregate process metrics: {}", e))
}

#[tauri::command]
pub async fn analytics_get_user_metrics(
    start_date: i64,
    end_date: i64,
    state: State<'_, AppDatabase>,
) -> Result<Vec<UserMetrics>, String> {
    let db = create_analytics_db_connection(&state)?;
    let aggregator = MetricsAggregator::new(db);

    aggregator
        .aggregate_by_user(start_date, end_date)
        .await
        .map_err(|e| format!("Failed to aggregate user metrics: {}", e))
}

#[tauri::command]
pub async fn analytics_get_tool_metrics(
    start_date: i64,
    end_date: i64,
    state: State<'_, AppDatabase>,
) -> Result<Vec<ToolMetrics>, String> {
    let db = create_analytics_db_connection(&state)?;
    let aggregator = MetricsAggregator::new(db);

    aggregator
        .aggregate_by_tool(start_date, end_date)
        .await
        .map_err(|e| format!("Failed to aggregate tool metrics: {}", e))
}

#[tauri::command]
pub async fn analytics_get_metric_trends(
    metric: String,
    days: usize,
    state: State<'_, AppDatabase>,
) -> Result<Vec<TrendPoint>, String> {
    let db = create_analytics_db_connection(&state)?;
    let aggregator = MetricsAggregator::new(db);

    aggregator
        .calculate_trends(&metric, days)
        .await
        .map_err(|e| format!("Failed to calculate trends: {}", e))
}

// AUDIT-ANALYTICS-087 fix: Aliases for specific trend commands
#[tauri::command]
pub async fn analytics_get_time_saved_trend(
    days: usize,
    state: State<'_, AppDatabase>,
) -> Result<Vec<TrendPoint>, String> {
    analytics_get_metric_trends("time_saved".to_string(), days, state).await
}

#[tauri::command]
pub async fn analytics_get_cost_saved_trend(
    days: usize,
    state: State<'_, AppDatabase>,
) -> Result<Vec<TrendPoint>, String> {
    analytics_get_metric_trends("cost_saved".to_string(), days, state).await
}

#[tauri::command]
pub async fn analytics_export_report(
    format: String,
    start_date: i64,
    end_date: i64,
    state: State<'_, AppDatabase>,
) -> Result<String, String> {
    let db = create_analytics_db_connection(&state)?;

    let calculator = ROICalculator::new(db.clone());
    let aggregator = MetricsAggregator::new(db.clone());
    let generator = ReportGenerator::new();

    let roi = calculator
        .calculate_roi(start_date, end_date)
        .await
        .map_err(|e| format!("Failed to calculate ROI: {}", e))?;

    match format.as_str() {
        "markdown" | "md" => {
            let process_metrics = aggregator
                .aggregate_by_process_type(start_date, end_date)
                .await
                .map_err(|e| format!("Failed to aggregate metrics: {}", e))?;
            Ok(generator.generate_executive_summary(&roi, &process_metrics))
        }
        "csv" => {
            let process_metrics = aggregator
                .aggregate_by_process_type(start_date, end_date)
                .await
                .map_err(|e| format!("Failed to aggregate metrics: {}", e))?;
            Ok(generator.generate_csv_export(&process_metrics))
        }
        "json" => {
            let process_metrics = aggregator
                .aggregate_by_process_type(start_date, end_date)
                .await
                .map_err(|e| format!("Failed to aggregate metrics: {}", e))?;
            let user_metrics = aggregator
                .aggregate_by_user(start_date, end_date)
                .await
                .map_err(|e| format!("Failed to aggregate metrics: {}", e))?;
            let tool_metrics = aggregator
                .aggregate_by_tool(start_date, end_date)
                .await
                .map_err(|e| format!("Failed to aggregate metrics: {}", e))?;

            generator
                .generate_json_export(&roi, &process_metrics, &user_metrics, &tool_metrics)
                .map_err(|e| format!("Failed to generate JSON: {}", e))
        }
        _ => Err(format!(
            "Unsupported format: {}. Use 'markdown', 'csv', or 'json'",
            format
        )),
    }
}

#[tauri::command]
pub async fn analytics_generate_weekly_report(
    user_id: String,
    state: State<'_, AppDatabase>,
) -> Result<String, String> {
    let db = create_analytics_db_connection(&state)?;
    let generator = ScheduledReportGenerator::new(db);

    generator.generate_weekly_report(&user_id).await
}

#[tauri::command]
pub async fn analytics_generate_monthly_report(
    user_id: String,
    state: State<'_, AppDatabase>,
) -> Result<String, String> {
    let db = create_analytics_db_connection(&state)?;
    let generator = ScheduledReportGenerator::new(db);

    generator.generate_monthly_report(&user_id).await
}

#[tauri::command]
pub async fn analytics_get_top_processes(
    start_date: i64,
    end_date: i64,
    limit: usize,
    state: State<'_, AppDatabase>,
) -> Result<Vec<ProcessMetrics>, String> {
    let db = create_analytics_db_connection(&state)?;
    let aggregator = MetricsAggregator::new(db);

    aggregator
        .get_top_processes(start_date, end_date, limit)
        .await
        .map_err(|e| format!("Failed to get top processes: {}", e))
}

#[tauri::command]
pub async fn analytics_save_snapshot(
    user_id: String,
    team_id: Option<String>,
    start_date: i64,
    end_date: i64,
    state: State<'_, AppDatabase>,
) -> Result<String, String> {
    let db = create_analytics_db_connection(&state)?;
    let calculator = ROICalculator::new(db);

    let roi = calculator
        .calculate_roi(start_date, end_date)
        .await
        .map_err(|e| format!("Failed to calculate ROI: {}", e))?;

    calculator
        .save_snapshot(&user_id, team_id.as_deref(), &roi)
        .await
        .map_err(|e| format!("Failed to save snapshot: {}", e))
}

#[tauri::command]
pub async fn track_workflow_view(
    workflow_id: String,
    user_id: String,
    state: State<'_, TelemetryState>,
) -> Result<(), String> {
    let collector = state.collector.read().await;

    if !collector.is_enabled() {
        return Ok(());
    }

    drop(collector);

    let mut properties = std::collections::HashMap::new();
    properties.insert("workflow_id".to_string(), serde_json::json!(workflow_id));
    properties.insert("user_id".to_string(), serde_json::json!(user_id));

    let event = TelemetryEvent {
        name: "workflow_view".to_string(),
        properties,
        timestamp: chrono::Utc::now().timestamp_millis() as u64,
        session_id: state.collector.read().await.get_session_id(),
        user_id: Some(user_id),
    };

    let collector = state.collector.write().await;
    collector
        .track(event)
        .await
        .map_err(|e| format!("Failed to track workflow view: {}", e))
}

#[tauri::command]
pub async fn acknowledge_milestone(
    milestone_id: String,
    user_id: String,
    state: State<'_, TelemetryState>,
) -> Result<(), String> {
    let collector = state.collector.read().await;

    if !collector.is_enabled() {
        return Ok(());
    }

    drop(collector);

    let mut properties = std::collections::HashMap::new();
    properties.insert("milestone_id".to_string(), serde_json::json!(milestone_id));
    properties.insert("user_id".to_string(), serde_json::json!(user_id));
    properties.insert(
        "acknowledged_at".to_string(),
        serde_json::json!(chrono::Utc::now().to_rfc3339()),
    );

    let event = TelemetryEvent {
        name: "milestone_acknowledged".to_string(),
        properties,
        timestamp: chrono::Utc::now().timestamp_millis() as u64,
        session_id: state.collector.read().await.get_session_id(),
        user_id: Some(user_id),
    };

    let collector = state.collector.write().await;
    collector
        .track(event)
        .await
        .map_err(|e| format!("Failed to acknowledge milestone: {}", e))
}

#[cfg(test)]
mod tests {
    use super::*;
    use crate::sys::telemetry::CollectorConfig;

    fn create_test_state() -> TelemetryState {
        let config = CollectorConfig {
            enabled: true,
            batch_size: 10,
            flush_interval_secs: 30,
            app_data_dir: None,
        };
        let collector = TelemetryCollector::new(config);
        let metrics_collector = AnalyticsMetricsCollector::new();
        TelemetryState::new(collector, metrics_collector)
    }

    #[tokio::test]
    async fn test_analytics_track_event() {
        let state = create_test_state();
        let event = TelemetryEvent {
            name: "test_event".to_string(),
            properties: HashMap::new(),
            timestamp: chrono::Utc::now().timestamp_millis() as u64,
            session_id: "test_session".to_string(),
            user_id: None,
        };

        let collector = state.collector.read().await;
        let result = collector.track(event).await;
        assert!(result.is_ok());
    }

    #[tokio::test]
    async fn test_analytics_get_session_id() {
        let state = create_test_state();
        let session_id = state.collector.read().await.get_session_id();
        assert!(!session_id.is_empty());
    }

    #[tokio::test]
    async fn test_metrics_get_system() {
        let state = create_test_state();
        let metrics = state
            .metrics_collector
            .write()
            .await
            .collect_system_metrics();
        assert!(metrics.memory_total_mb > 0);
    }

    #[tokio::test]
    async fn test_metrics_get_app() {
        let state = create_test_state();
        let metrics = state.metrics_collector.read().await.collect_app_metrics();
        assert_eq!(metrics.automations_count, 0);
        assert_eq!(metrics.goals_count, 0);
    }

    #[tokio::test]
    async fn test_feature_flag_get() {
        let state = SettingsState::new();
        // Insert parallel_execution = true
        {
            let mut settings = state.settings.lock().await;
            settings
                .feature_flags
                .insert("parallel_execution".to_string(), true);
        }

        let result = get_feature_flag("parallel_execution", &state).await;
        assert!(result.is_ok());
        assert!(result.unwrap());

        let result = get_feature_flag("unknown_flag", &state).await;
        assert!(result.is_ok());
        assert!(!result.unwrap());
    }

    #[tokio::test]
    async fn test_feature_flag_get_all() {
        let state = SettingsState::new();
        {
            let mut settings = state.settings.lock().await;
            settings
                .feature_flags
                .insert("parallel_execution".to_string(), true);
        }

        let result = get_all_feature_flags(&state).await;
        assert!(result.is_ok());

        let flags = result.unwrap();
        assert!(!flags.is_empty());
        assert_eq!(flags.get("parallel_execution"), Some(&true));
    }
}

File: packages/api/src/analytics.ts
/**
 * Analytics & Metrics API — typed wrappers for analytics_*, metrics_*, and feature_flag_* commands.
 */

import { command } from '@agiworkforce/runtime';

// ---- Types ----

export interface TelemetryEvent {
  name: string;
  properties?: Record<string, unknown>;
}
export interface SystemMetrics {
  cpuUsage: number;
  memoryUsage: number;
  diskUsage: number;
  uptime: number;
}
export interface AppMetrics {
  activeConversations: number;
  totalMessages: number;
  automationsRun: number;
  mcpServers: number;
  cacheHitRate: number;
}
export interface ROIReport {
  timeSavedHours: number;
  costSavedUsd: number;
  automationsRun: number;
  processesOptimized: number;
}
export interface ProcessMetrics {
  name: string;
  runs: number;
  avgDuration: number;
  successRate: number;
  timeSaved: number;
}
export interface UserMetrics {
  userId: string;
  actions: number;
  automationsRun: number;
  timeSaved: number;
}
export interface ToolMetrics {
  toolName: string;
  calls: number;
  avgDuration: number;
  successRate: number;
}
export interface TrendPoint {
  timestamp: number;
  value: number;
}
export interface RealtimeStats {
  activeUsers: number;
  runningAutomations: number;
  pendingTasks: number;
  messagesPerMinute: number;
}
export interface RecordAutomationRequest {
  automationType: string;
  durationMs: number;
  success: boolean;
  [key: string]: unknown;
}
export interface MetricsSnapshot {
  timestamp: string;
  automations: number;
  timeSaved: number;
  costSaved: number;
}
export interface Comparison {
  manual: number;
  automated: number;
  savings: number;
}
export interface PeriodComparison {
  current: MetricsSnapshot;
  previous: MetricsSnapshot;
  changePercent: number;
}
export interface BenchmarkComparison {
  yours: number;
  industry: number;
  percentile: number;
}
export interface MilestoneData {
  id: string;
  name: string;
  achieved: boolean;
  achievedAt?: string;
}
export interface DayStats {
  automations: number;
  timeSaved: number;
  messages: number;
}
export interface WeekStats {
  automations: number;
  timeSaved: number;
  messages: number;
  trend: number;
}
export interface MonthStats {
  automations: number;
  timeSaved: number;
  costSaved: number;
}
export interface AllTimeStats {
  automations: number;
  timeSaved: number;
  costSaved: number;
  messages: number;
}
export interface ComparisonData {
  manual: number;
  automated: number;
  savings: number;
}
export interface PeriodComparisonData {
  current: number;
  previous: number;
  change: number;
}
export interface BenchmarkComparisonData {
  yours: number;
  benchmark: number;
  percentile: number;
}
export interface ActivityItem {
  id: string;
  type: string;
  description: string;
  timestamp: string;
}
export interface ExportOptions {
  format: string;
  startDate?: number;
  endDate?: number;
}

// ---- Analytics ----

export async function analyticsTrackEvent(event: TelemetryEvent): Promise<void> {
  return command<void>('analytics_track_event', { event });
}
export async function analyticsFlushEvents(): Promise<void> {
  return command<void>('analytics_flush_events');
}
export async function analyticsGetSessionId(): Promise<string> {
  return command<string>('analytics_get_session_id');
}
export async function analyticsSetUserProperty(key: string, value: unknown): Promise<void> {
  return command<void>('analytics_set_user_property', { key, value });
}
export async function analyticsDeleteAllData(): Promise<void> {
  return command<void>('analytics_delete_all_data');
}
export async function analyticsGetUsageStats(): Promise<unknown> {
  return command<unknown>('analytics_get_usage_stats');
}
export async function analyticsGetFeatureUsage(): Promise<unknown[]> {
  return command<unknown[]>('analytics_get_feature_usage');
}
export async function analyticsCalculateRoi(
  startDate: number,
  endDate: number,
): Promise<ROIReport> {
  return command<ROIReport>('analytics_calculate_roi', { startDate, endDate });
}
export async function analyticsGetProcessMetrics(
  startDate: number,
  endDate: number,
): Promise<ProcessMetrics[]> {
  return command<ProcessMetrics[]>('analytics_get_process_metrics', { startDate, endDate });
}
export async function analyticsGetUserMetrics(
  startDate: number,
  endDate: number,
): Promise<UserMetrics[]> {
  return command<UserMetrics[]>('analytics_get_user_metrics', { startDate, endDate });
}
export async function analyticsGetToolMetrics(
  startDate: number,
  endDate: number,
): Promise<ToolMetrics[]> {
  return command<ToolMetrics[]>('analytics_get_tool_metrics', { startDate, endDate });
}
export async function analyticsGetMetricTrends(
  metric: string,
  days: number,
): Promise<TrendPoint[]> {
  return command<TrendPoint[]>('analytics_get_metric_trends', { metric, days });
}
export async function analyticsGetTimeSavedTrend(days: number): Promise<TrendPoint[]> {
  return command<TrendPoint[]>('analytics_get_time_saved_trend', { days });
}
export async function analyticsGetCostSavedTrend(days: number): Promise<TrendPoint[]> {
  return command<TrendPoint[]>('analytics_get_cost_saved_trend', { days });
}
export async function analyticsExportReport(
  format: string,
  startDate: number,
  endDate: number,
): Promise<string> {
  return command<string>('analytics_export_report', { format, startDate, endDate });
}
export async function analyticsGenerateWeeklyReport(): Promise<string> {
  return command<string>('analytics_generate_weekly_report');
}
export async function analyticsGenerateMonthlyReport(): Promise<string> {
  return command<string>('analytics_generate_monthly_report');
}
export async function analyticsGetTopProcesses(
  startDate: number,
  endDate: number,
  limit: number,
): Promise<ProcessMetrics[]> {
  return command<ProcessMetrics[]>('analytics_get_top_processes', { startDate, endDate, limit });
}
export async function analyticsSaveSnapshot(
  startDate: number,
  endDate: number,
  teamId?: string,
): Promise<string> {
  return command<string>('analytics_save_snapshot', { teamId, startDate, endDate });
}
export async function trackWorkflowView(workflowId: string): Promise<void> {
  return command<void>('track_workflow_view', { workflowId });
}
export async function acknowledgeMilestone(milestoneId: string): Promise<void> {
  return command<void>('acknowledge_milestone', { milestoneId });
}

// ---- System/App Metrics ----

export async function metricsGetSystem(): Promise<SystemMetrics> {
  return command<SystemMetrics>('metrics_get_system');
}
export async function metricsGetApp(): Promise<AppMetrics> {
  return command<AppMetrics>('metrics_get_app');
}
export async function metricsIncrementAutomations(): Promise<void> {
  return command<void>('metrics_increment_automations');
}
export async function metricsIncrementGoals(): Promise<void> {
  return command<void>('metrics_increment_goals');
}
export async function metricsSetMcpServers(count: number): Promise<void> {
  return command<void>('metrics_set_mcp_servers', { count });
}
export async function metricsSetCacheHitRate(rate: number): Promise<void> {
  return command<void>('metrics_set_cache_hit_rate', { rate });
}

// ---- Feature Flags ----

export async function featureFlagGet(flagName: string): Promise<boolean> {
  return command<boolean>('feature_flag_get', { flagName });
}
export async function featureFlagGetAll(): Promise<Record<string, boolean>> {
  return command<Record<string, boolean>>('feature_flag_get_all');
}

// ---- ROI / Comparison Metrics ----

export async function getRealtimeStats(): Promise<RealtimeStats> {
  return command<RealtimeStats>('get_realtime_stats');
}
export async function recordAutomationMetrics(
  request: RecordAutomationRequest,
): Promise<MetricsSnapshot> {
  return command<MetricsSnapshot>('record_automation_metrics', { request });
}
export async function getMetricsHistory(days: number): Promise<MetricsSnapshot[]> {
  return command<MetricsSnapshot[]>('get_metrics_history', { days });
}
export async function compareToManual(automationType: string): Promise<Comparison> {
  return command<Comparison>('compare_to_manual', { automationType });
}
export async function compareToPreviousPeriod(days: number): Promise<PeriodComparison> {
  return command<PeriodComparison>('compare_to_previous_period', { days });
}
export async function compareToIndustryBenchmark(role: string): Promise<BenchmarkComparison> {
  return command<BenchmarkComparison>('compare_to_industry_benchmark', { role });
}
export async function getMilestones(): Promise<MilestoneData[]> {
  return command<MilestoneData[]>('get_milestones');
}
export async function shareMilestone(milestoneId: string): Promise<void> {
  return command<void>('share_milestone', { milestoneId });
}
export async function getTodayStats(): Promise<DayStats> {
  return command<DayStats>('get_today_stats');
}
export async function getWeekStats(): Promise<WeekStats> {
  return command<WeekStats>('get_week_stats');
}
export async function getMonthStats(): Promise<MonthStats> {
  return command<MonthStats>('get_month_stats');
}
export async function getAllTimeStats(): Promise<AllTimeStats> {
  return command<AllTimeStats>('get_all_time_stats');
}
export async function getManualVsAutomatedComparison(
  automationType: string,
): Promise<ComparisonData> {
  return command<ComparisonData>('get_manual_vs_automated_comparison', { automationType });
}
export async function getPeriodComparison(period: string): Promise<PeriodComparisonData> {
  return command<PeriodComparisonData>('get_period_comparison', { period });
}
export async function getBenchmarkComparison(role: string): Promise<BenchmarkComparisonData> {
  return command<BenchmarkComparisonData>('get_benchmark_comparison', { role });
}
export async function getRecentActivity(limit: number): Promise<ActivityItem[]> {
  return command<ActivityItem[]>('get_recent_activity', { limit });
}
export async function exportRoiReport(options: ExportOptions): Promise<string> {
  return command<string>('export_roi_report', { options });
}
